packages/number-field/src/NumberField.ts

```diff
@@ -496,6 +496,18 @@ export class NumberField extends TextfieldBase {
 
     protected override handleInput(event: InputEvent): void {
         if (this.isComposing) {
+            // If user actually types a new character.
+            if (event.data) {
+                // Don't allow non-numeric characters even in composing mode.
+                const partialValue = this.convertValueToNumber(event.data);
+
+                if (Number.isNaN(partialValue)) {
+                    this.inputElement.value = this.indeterminate
+                        ? indeterminatePlaceholder
+                        : this._trackingValue;
```

Comment:
Resetting the value to `this._trackingValue` can be confusing! Can we show some validation feedback instead of a reset?

Reply:
I followed the same logic as on english keyboard (line 547), we are resetting here because you should not be able to type non-numeric numbers, so we're removing what you typed.

Reply:
Makes Sense!

Reply:
good catch, fixed 👍  can you try again?